django/db/models/expressions.py

```diff
@@ -483,7 +485,14 @@ def __init__(self, value, output_field=None):
         self.value = value
 
     def as_sql(self, compiler, connection):
-        return '%s', [self.value]
+        val = self.value
+        if self._output_field_or_none is not None:
+            from django.db.models.sql.compiler import SQLUpdateCompiler
+            if isinstance(compiler, SQLUpdateCompiler):
```

Comment:
Yeah, I don't like it either. That's why I mentioned it in my initial comment.
```SQLUpdateCompiler``` calls ```field.get_db_prep_save(val)``` on raw values, but it's not happening with Value objects.
So ```qs.update(field=value)``` works but ```qs.update(field=Value(value))``` fails for ```Field```s that require conversion. I've added test cases to ```tests.expressions``` that fail without this.
```UUIDField``` is problematic for MySQL.
```UUIDField```, ```TimeField``` and ```DurationField``` are problematic for SQLite.
I did the compiler type check here, since I'm not sure values should be processed with ```get_db_prep_save``` in the other compiler types.